src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobieUserRepositoryInterpreter.scala

```diff
@@ -43,9 +48,20 @@ class DoobieUserRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
 
   import UserSQL._
 
-  def put(user: User): F[User] =
+  def create(user: User): F[User] =
     insert(user).withUniqueGeneratedKeys[Long]("ID").map(id => user.copy(id = id.some)).transact(xa)
 
+  def update(user: User): F[Option[User]] = {
+    val existingUser : ConnectionIO[Option[User]] = user.id.flatTraverse(select(_).option)
```

Comment:
this is a little confusing.  If the user id is not present, we should just return F(None).

Seems like we are doing a lot of gymnastics in there.  Would something like this suffice?

```
  def update(user: User): F[Option[User]] = {
    OptionT.fromOption(user.id).semiflatMap(id => UserSQL.update(user, id).run.transact(xa).map(_ => user)).value
  }
```

Reply:
Yes I think you and I were fiddling with the same thing for the last few minutes. We do have to include the explicit type parameter `[F]` there.

Reply:
I came up with this:

```
  def update(user: User): F[Option[User]] = OptionT.fromOption[ConnectionIO](user.id).semiflatMap { id =>
    UserSQL.update(user, id).run
  }.value.map(_ => user.some).transact(xa)
```

For some reason I think it's nice to keep the transact call at the end.

Reply:
word.  Can we use `as` instead of `map` here?

Reply:
Yeah, I'm going to rebase on top of the one you merged, and include that then